repl/src/main/scala/ammonite/repl/interp/Interpreter.scala

```diff
@@ -94,46 +103,52 @@ class Interpreter(prompt0: Ref[String],
     replApi
   )
   Timer("Interpreter init eval")
-  init()
-  Timer("Interpreter init init")
+
   val argString = replArgs.zipWithIndex.map{ case (b, idx) =>
     s"""
     val ${b.name} =
       ammonite.repl.frontend.ReplBridge.repl.replArgs($idx).value.asInstanceOf[${b.typeTag.tpe}]
     """
   }.mkString("\n")
 
-  val predefs = Seq(
-    (hardcodedPredef, Name("HardcodedPredef")),
-    (predef, Name("Predef")),
-    (storage.loadPredef, Name("LoadedPredef")),
-    (argString, Name("ArgsPredef"))
-  )
```

Comment:
The changes to this block and the one below it seem unnecessary; if the only thing you wanted to do is add `if !sourceCode.isEmpty` to the for loop, and change `processModule0` to `processModule`, you can do so without changing this whole block.

Also, you should not need to do the `if !sourceCode.isEmpty` check. If your caching is working properly, it would be cached on first run and not actually make things faster. All it does is mask deeper problems, like the problems with a comment-only predef I found earlier

Also